src/form/ivr.jl

```diff
@@ -0,0 +1,694 @@
+    variable_mc_transformer_current_real(pm, nw=nw, bounded=bounded, report=report; kwargs...)
+    variable_mc_transformer_current_imaginary(pm, nw=nw, bounded=bounded, report=report; kwargs...)
+
+    # store expressions in rectangular power variable space
+    p = Dict()
+    q = Dict()
+
+    f_cnd = [1, 2, 3] #TODO extend to this being passed in constraint template
+    t_cnd = [1, 2, 3] #TODO extend to this being passed in constraint template
+
+    for (l,i,j) in _PMs.ref(pm, nw, :arcs_from_trans)
+        vr_fr = _PMs.var(pm, nw, :vr, i)[f_cnd]
+        vi_fr = _PMs.var(pm, nw, :vi, i)[f_cnd]
+        cr_fr = _PMs.var(pm, nw, :crt, (l,i,j))
+        ci_fr = _PMs.var(pm, nw, :cit, (l,i,j))
+
+        vr_to = _PMs.var(pm, nw, :vr, j)[t_cnd]
+        vi_to = _PMs.var(pm, nw, :vi, j)[t_cnd]
+        cr_to = _PMs.var(pm, nw, :crt, (l,j,i))
+        ci_to = _PMs.var(pm, nw, :cit, (l,j,i))
+        p[(l,i,j)] = vr_fr.*cr_fr  + vi_fr.*ci_fr
+        q[(l,i,j)] = vi_fr.*cr_fr  - vr_fr.*ci_fr
+        p[(l,j,i)] = vr_to.*cr_to  + vi_to.*ci_to
+        q[(l,j,i)] = vi_to.*cr_to  - vr_to.*ci_to
+    end
+
+    _PMs.var(pm, nw)[:p] = p
+    _PMs.var(pm, nw)[:q] = q
+    report && _PMs.sol_component_value_edge(pm, nw, :transformer, :pf, :pt, _PMs.ref(pm, nw, :arcs_from_trans), _PMs.ref(pm, nw, :arcs_to_trans), p)
+    report && _PMs.sol_component_value_edge(pm, nw, :transformer, :qf, :qt, _PMs.ref(pm, nw, :arcs_from_trans), _PMs.ref(pm, nw, :arcs_to_trans), q)
+end
+
+
+""
+function variable_mc_load(pm::_PMs.AbstractIVRModel; nw::Int=pm.cnw, bounded::Bool=true, report::Bool=true, kwargs...)
+    _PMs.var(pm, nw)[:crd] = Dict{Int, Any}()
+    _PMs.var(pm, nw)[:cid] = Dict{Int, Any}()
+    _PMs.var(pm, nw)[:crd_bus] = Dict{Int, Any}()
+    _PMs.var(pm, nw)[:cid_bus] = Dict{Int, Any}()
+end
+
+""
+function variable_mc_generation(pm::_PMs.AbstractIVRModel; nw::Int=pm.cnw, bounded::Bool=true, report::Bool=true, kwargs...)
+    variable_mc_generation_current_real(pm, nw=nw, bounded=bounded, report=report; kwargs...)
+    variable_mc_generation_current_imaginary(pm, nw=nw, bounded=bounded, report=report; kwargs...)
+
+    cnds = _PMs.conductor_ids(pm; nw=nw)
+    ncnds = length(cnds)
+
+    _PMs.var(pm, nw)[:crg_bus] = Dict{Int, Any}()
+    _PMs.var(pm, nw)[:cig_bus] = Dict{Int, Any}()
+
+    # store active and reactive power expressions for use in objective + post processing
+    _PMs.var(pm, nw)[:pg] = Dict{Int, Any}()
+    _PMs.var(pm, nw)[:qg] = Dict{Int, Any}()
+
+    # report && _PMs.sol_component_value(pm, nw, :gen, :pg, _PMs.ids(pm, nw, :gen), pg)
+    # report && _PMs.sol_component_value(pm, nw, :gen, :qg, _PMs.ids(pm, nw, :gen), qg)
+
+    # if bounded
+    #     for (i,gen) in _PMs.ref(pm, nw, :gen)
+    #         constraint_mc_generation_active_power_limits(pm, i, nw=nw)
+    #         constraint_mc_generation_reactive_power_limits(pm, i, nw=nw)
+    #     end
+    # end
```

Comment:
reason commented out instead of removed?

Reply:
This is leftover from Frederik's implementation. I moved these limits to the constraints instead, because it has to be specialized based on whether it is delta or not. I will remove these lines.